miso-web/src/main/webapp/scripts/hot_library.js

```diff
@@ -73,6 +73,57 @@ HotTarget.library = (function() {
     };
   };
 
+  var checkQcs = function(table) {
+    Utils.showDialog('QC Criteria', 'Check', [{
+      label: 'Concentration',
+      type: 'compare',
+      compareTypeProperty: 'concentrationCompareType',
+      valueProperty: 'concentrationCompareValue'
+    }, {
+      label: 'Volume',
+      type: 'compare',
+      compareTypeProperty: 'volumeCompareType',
+      valueProperty: 'volumeCompareValue'
+    }, {
+      label: 'Size',
+      type: 'compare',
+      compareTypeProperty: 'sizeCompareType',
+      valueProperty: 'sizeCompareValue'
+    }], function(output) {
+      var rowCount = table.countRows();
+      var changes = [];
+      for (var row = 0; row < rowCount; row++) {
+        var concentration = table.getDataAtRowProp(row, 'concentration');
+        var volume = table.getDataAtRowProp(row, 'volume');
+        var size = table.getDataAtRowProp(row, 'dnaSize');
+        var pass = checkQc(concentration, output.concentrationCompareType, output.concentrationCompareValue)
+            && checkQc(volume, output.volumeCompareType, output.volumeCompareValue)
+            && checkQc(size, output.sizeCompareType, output.sizeCompareValue);
+        changes.push([row, 'qcPassed', pass ? 'True' : 'False']);
+      }
+      table.setDataAtRowProp(changes);
+    });
+  };
+
+  var checkQc = function(value, compareType, compareValue) {
```

Comment:
I don't mind this, but it might be simpler to return a function that does the comparison. Rather than having two properties, return a single property that's `function(x)`. For the first dropdown, have that match an array of `function(x, y)` and then when the user changes the text box:
```
var comparator =  comparators[select.value];
var value = parseInt(valueControl.value);
result[property] = function(x) { return comparator(x , value); };
```